docs/UsersGuide/source/ContributorsGuide.rst

```diff
@@ -0,0 +1,369 @@
+    * If possible, developers should run a fundamental test suite (see :numref:`Section %s <Testing>`) on at least one supported platform and report on the outcome in the PR template.
+    * If changes are made to ``regional_workflow``, a corresponding PR to ``ufs-srweather-app`` should be opened to update the regional_workflow hash. 
+    * Update the ``.rst`` documentation files where appropriate. 
+
+..
+    COMMENT: Is this something contributors need to do, or code managers only?
+
+
+**Guidelines for New Components:**
+    * Components should have a mechanism for portability and platform-independence; code that is included in the SRW App should not be tied to specific platforms. 
+    * New components should be able to build using the standard supported NCEPLIBS environment (currently `HPC-Stack <https://github.com/NOAA-EMC/hpc-stack>`__).
+    * New entries in Externals.cfg should only be repositories from “official” sources; either the `UFS Community GitHub organization <https://github.com/ufs-community>`__ or another NOAA project organization.
+    * Document how to use the new component, and include this information in the PR. 
+
+
+.. _Testing: 
+
+Testing
+===============
+
+The ``ufs-srweather-app`` repository uses the established workflow end-to-end (WE2E) testing framework (see :numref:`Chapter %s <WE2E_tests>`) to implement two tiers of testing: fundamental and comprehensive. *Fundamental testing* consists of a lightweight set of tests that can be automated and run regularly on each `Level 1 <https://github.com/ufs-community/ufs-srweather-app/wiki/Supported-Platforms-and-Compilers>`__ platform. These are mostly low-resolution tests and cover a wide scope of capabilities to ensure that there are no major, obvious faults in the underlying code. *Comprehensive testing* includes the entire set of WE2E tests. 
+
+Before opening a PR, a minimum set of tests should be run: 
+    * At least one end-to-end test (preferably a fundamental test suite) should be run on at least one supported platform
+    * Any new functionality should be tested explicitly, and tests should be described in detail in the PR message. Depending on the impact of this functionality, this test should be added to the WE2E suite of fundamental or comprehensive tests. 
+
+..
+    COMMENT: By "supported platforms," are we talking about Level 1 only?
+
+**Updating the Testing Suite:** When new capabilities are added or new bugs/issues are discovered, tests should be created and/or modified to test for these conditions. For example, if a new physics suite is introduced, it may be possible to alter an existing test rather than creating an entirely new test. Code developers introducing new capabilities should work with code managers to provide the proper configuration files, data, and other information necessary to create new tests for these capabilities.
```

Comment:
"When new capabilities are added or new bugs/issues are discovered, tests should be created and/or modified to test for these conditions." -> "When new capabilities are added or new bugs/issues are discovered, WE2E tests should be created and/or modified to verify new features are not broken in subsequent PRs."